docs/src/main/asciidoc/_high-availability/subsystem-support/Distributable_Web_Applications.adoc

```diff
@@ -0,0 +1,154 @@
+If undefined, the default cache of the associated cache container will be used.
+granularity::
+This defines how the session manager will map a session into individual cache entries.
+Possible values are:
+SESSION:::
+Stores all session attributes within a single cache entry.
+This is generally more expensive than ATTRIBUTE granularity, but preserves any cross-attribute object references.
+ATTRIBUTE:::
+Stores each session attribute within a separate cache entry.
+This is generally more efficient than SESSION granularity, but does not preserve any cross-attribute object references.
+
+affinity::
+This resource defines the affinity that a web request should have for a given server.
+The affinity of the associated web session determines the algorithm for generating the route to be appended onto the session ID (within the JSESSIONID cookie, or when encoding URLs).
+This annotation of the session ID is used by load balancers to advise how future requests for existing sessions should be directed.
+Routing is designed to be opaque to application code such that calls to `HttpSession.getId()` always return an unmodified session ID.
+This is only generated when creating the JSESSIONID cookie, or when encoding URLs via `HttpServletResponse.encodeURL()` and `encodeRedirectURL()`.
+Possible values are:
+affinity=none:::
+Web requests will have no affinity to any particular node.
+This option is intended for use cases where web session state is not maintained within the application server.
+affinity=local:::
+Web requests will have an affinity to the server that last handled a request for a given session.
+This option corresponds to traditional sticky session behavior.
+affinity=primary-owner:::
+Web requests will have an affinity to the primary owner of a given session.
+Behaves the same as affinity=local if the backing cache is not distributed nor replicated.
+
+== Overriding default behavior
+
+A web application can override the default distributable session management behavior in 1 of 2 ways:
+
+1. Reference a session-management profile by name
+1. Provide deployment-specific session management configuration
+
+=== Referencing an existing session management profile
+
+To use an existing session management profile, a web application should include a distributable-web.xml deployment descriptor located within the application's /WEB-INF directory.
+
+e.g.
+
+./WEB-INF/distributable-web.xml
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<distributable-web xmlns="urn:jboss:distributable-web:1.0">
+    <session-management name="foo"/>
+</distributable-web>
+----
+
+Alternatively, the same XML can be defined within an existing jboss-all.xml deployment descriptor:
+
+e.g.
+
+./META-INF/jboss-all.xml
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<jboss xmlns="urn:jboss:1.0">
+    <distributable-web xmlns="urn:jboss:distributable-web:1.0">
+        <session-management name="foo"/>
+    </distributable-web>
+</jboss>
+----
+
```

Comment:
@pferraro what about adding some cli scripts? :

The "foo" session management profile can be created on the server like this:
----
/subsystem=distributable-web/infinispan-session-management=sm_offload:add(cache-container=web,cache=foo,granularity=SESSION)
----
Note that a cache named "foo" in cache container "web" must exist in subsystem "infinispan"; if it did not exist already, it can be created like this:
----
/subsystem=infinispan/cache-container=web/distributed-cache=foo:add()
----

for "affinity=none":
----
/subsystem=distributable-web/infinispan-session-management=sm_offload/affinity=none:add()
----

for "affinity=local":
----
/subsystem=distributable-web/infinispan-session-management=sm_offload/affinity=local:add()
----

for "affinity=primary-owner":
----
/subsystem=distributable-web/infinispan-session-management=sm_offload/affinity=primary-owner:add()
----